src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,113 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const {
+    title, description, imgUrl, imdbUrl, imdbId,
+  } = newMovie;
+
+  const handleInputChange = (newValue: string, name: string) => {
+    setNewMovie((prevMovie) => ({
+      ...prevMovie,
+      [name]: newValue,
+    }));
+  };
+
+  const movie = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
+
+  // eslint-disable-next-line max-len
+  const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+  const emptyFields = title.trim() && imgUrl.match(pattern)
+  && imdbUrl.match(pattern) && imdbId.trim();
+
+  const reset = () => {
+    setNewMovie(movie);
+  };
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+
+    if (!emptyFields) {
+      return;
+    }
+
+    onAdd(movie);
```

Comment:
```suggestion
    onAdd(newMovie);
```
i think it should be newMovie state instead of the empty object